daffodil-runtime2/src/main/scala/org/apache/daffodil/runtime2/Runtime2DataProcessor.scala

```diff
@@ -188,11 +196,13 @@ final class ParseResult(outfile: os.Path,
 }
 
 final class UnparseResult(val finalBitPos0b: Long,
-                          override val processorStatus: ProcessorResult,
-                          loc: DataLocation = Runtime2DataLocation())
+                          override val processorStatus: ProcessorResult)
   extends DFDL.UnparseResult
     with DFDL.State
     with WithDiagnosticsImpl {
+
+  val loc: DataLocation = Runtime2DataLocation(finalBitPos0b + 1, (finalBitPos0b + 1) / 8)
+
   /**
```

Comment:
Thanks for the heads up and fixing those locations.  You're setting them to the end of the input (why add 1, though?).

Reply:
comment added. Because DataLocation is 1-based numbers.